extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -409,6 +409,11 @@ export class AmpAnalytics extends AMP.BaseElement {
    * @return {boolean} true if the user has opted out.
    */
   hasOptedOut_() {
+    const elementId = this.config_['optoutElementId'];
+    if (elementId && this.win.document.getElementById(elementId)) {
```

Comment:
What happens for the FIE case? When the element is inserted into the parent doc, do we want to opt out analytics within the ad?

Reply:
Not sure if I'm misinterpreting your question, but the optout is for Google Analytics and it isn't part of an ad. Optout would not touch anything related to ads. The optout would happen on the website owner's page to prevent Google Analytics collect hits.

Reply:
@leeandrew1693 
I am asking about the case where an AMPHTML ad is served on an AMP page. In that case there will be only one single AMP runtime where we can choose to also opt out google analytics within the ad. The same applies to shadow doc.  
I guess a good question to ask is what happens in non AMP case? Does the optout applies only to the top parent page, or to all googleanalytics usage (within iframe, shadow doc...) Thank you

Reply:
Optout applies to all google analytics usage, including within iframes. We've configured the extension so that it will inject the optout element/script into every page and every iframe.

Reply:
@zhouyx Does that clarify your questions?

Reply:
Thank you @leeandrew1693 and sorry for the late response. 
>We've configured the extension so that it will inject the optout element/script into every page and every iframe.

I see. That makes sense. And given we are checking the element in the window document, it should also work for the shadow doc case. LGTM